src/oopFundamentals/Kata6/Person.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace src\oopFundamentals\Kata6;
+
+class Person
+{
+    public const SPECIES = 'Homo Sapiens';
+
+    public const AGE_VALIDATION_ERROR_MESSAGE = 'Age must be a non-negative integer!';
+
+    public const OCCUPATION_VALIDATION_ERROR = 'Occupation must be a string!';
+
+    public function __construct(
+        protected int $age,
+        protected string $name,
+        protected $occupation,
+    ) {
+        if ($this->age < 0) {
+            throw new \InvalidArgumentException(self::AGE_VALIDATION_ERROR_MESSAGE);
+        }
+
+        if (gettype($this->occupation) !== 'string') {
+            throw new \InvalidArgumentException(self::OCCUPATION_VALIDATION_ERROR);
+        }
+    }
+
+    public function getAge(): int
+    {
+        return $this->age;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getOccupation(): string
+    {
+        return $this->occupation;
+    }
+
+    public function setAge(int $age): self
```

Comment:
I guess, you'll miss this Kata requirements to throw the same message as in constructor, if you use typed argument here.
It's applicable to every Set methods in this file.

Reply:
Updated, thanks!